vnext/local-cli/config/configUtils.js

```diff
@@ -0,0 +1,259 @@
+ */
+function findDependencyProjectFiles(winFolder) {
+  // First, search for all potential project files
+  const allCppProj = findFiles(winFolder, '*.vcxproj');
+  const allCsProj = findFiles(winFolder, '*.csproj');
+
+  const allProjects = allCppProj.concat(allCsProj);
+
+  if (allProjects.length === 0) {
+    // If there're no project files, return 0
+    return [];
+  }
+
+  var dependencyProjectFiles = [];
+
+  // Try to find any project file that appears to be a dependency project
+  for (const projectFile of allProjects) {
+    if (isRnwDependencyProject(path.join(winFolder, projectFile))) {
+      dependencyProjectFiles.push(projectFile);
+    }
+  }
+
+  return dependencyProjectFiles;
+}
+
+/**
+ * Checks if the target file path is a RNW app project file.
+ * @param {string} filePath The absolute file path to check.
+ * @return {boolean} Whether the path is to a RNW app project file.
+ */
+function isRnwAppProject(filePath) {
+  return (
+    fs
+      .readFileSync(filePath)
+      .toString()
+      .search(/Microsoft\.ReactNative\.Uwp\.(Cpp|CSharp)App\.targets/) > 0
+  );
+}
+
+/**
+ * Search for the RNW app project files under the target folder.
+ * @param {string} winFolder The absolute path to target folder.
+ * @return {array} Return the array of relative file paths.
+ */
+function findAppProjectFiles(winFolder) {
+  // First, search for all potential project files
+  const allCppProj = findFiles(winFolder, '*.vcxproj');
+  const allCsProj = findFiles(winFolder, '*.csproj');
+
+  const allProjects = allCppProj.concat(allCsProj);
+
+  if (allProjects.length === 0) {
+    // If there're no project files, return 0
+    return [];
+  }
+
+  var appProjectFiles = [];
+
+  // Try to find any project file that appears to be an app project
+  for (const projectFile of allProjects) {
+    if (isRnwAppProject(path.join(winFolder, projectFile))) {
+      appProjectFiles.push(projectFile);
+    }
+  }
+
+  return appProjectFiles;
+}
+
+/**
+ * Returns the programming language of the project file.
+ * @param {string} projectPath The project file path to check.
+ * @return {string} The language string: cpp, cs, or null if unknown.
+ */
+function getProjectLanguage(projectPath) {
+  if (projectPath.endsWith('.vcxproj')) {
+    return 'cpp';
+  } else if (projectPath.endsWith('.csproj')) {
+    return 'cs';
+  }
+  return null;
+}
+
+/**
+ * Reads in the contents of the target project file.
+ * @param {string} projectPath The target project file path.
+ * @return {string} The project file contents.
+ */
+function readProjectFile(projectPath) {
+  return fs.readFileSync(projectPath, 'utf8').toString();
+}
+
+/**
+ * Search for the given XML tag in the project contents and return its value.
+ * @param {string} projectContents The XML project contents.
+ * @param {string} tagName The XML tag to look for.
+ * @return {string} The value of the tag if it exists.
+ */
+function findTagValue(projectContents, tagName) {
```

Comment:
can we just use JavaScript's XML parsing instead? this would break if we ever added even a space or a condition.
For example:
```xml
<MyTag >value</ MyTag>
```
or
```xml
<MyTag Condition="foo">value</MyTag>
```
would fail. Using JavaScript's XML parsing would let us at least deal with the first case, although the 2nd case would need more work/integration with MSBuild that's probably hard to make happen.

Reply:
Is there one that we're already using? I don't want to add a new xml parser dependency to our repo. There's like a million XML parsers, and the one we were including as a dependency before was junk.

Reply:
Switching to using `xmldom` and `xpath`, thouugh that means adding a dependency.